moveit_core/collision_detection_bullet/include/moveit/collision_detection_bullet/collision_robot_bt.h

```diff
@@ -0,0 +1,100 @@
+                          const robot_state::RobotState& state2) const override;
+  void checkSelfCollision(const CollisionRequest& req, CollisionResult& res, const robot_state::RobotState& state1,
+                          const robot_state::RobotState& state2, const AllowedCollisionMatrix& acm) const override;
+
+  void checkOtherCollision(const CollisionRequest& req, CollisionResult& res, const robot_state::RobotState& state,
+                           const CollisionRobot& other_robot,
+                           const robot_state::RobotState& other_state) const override;
+  void checkOtherCollision(const CollisionRequest& req, CollisionResult& res, const robot_state::RobotState& state,
+                           const CollisionRobot& other_robot, const robot_state::RobotState& other_state,
+                           const AllowedCollisionMatrix& acm) const override;
+  void checkOtherCollision(const CollisionRequest& req, CollisionResult& res, const robot_state::RobotState& state1,
+                           const robot_state::RobotState& state2, const CollisionRobot& other_robot,
+                           const robot_state::RobotState& other_state1,
+                           const robot_state::RobotState& other_state2) const override;
+  void checkOtherCollision(const CollisionRequest& req, CollisionResult& res, const robot_state::RobotState& state1,
+                           const robot_state::RobotState& state2, const CollisionRobot& other_robot,
+                           const robot_state::RobotState& other_state1, const robot_state::RobotState& other_state2,
+                           const AllowedCollisionMatrix& acm) const override;
+
+  void distanceSelf(const DistanceRequest& req, DistanceResult& res,
+                    const robot_state::RobotState& state) const override;
+
+  void distanceOther(const DistanceRequest& req, DistanceResult& res, const robot_state::RobotState& state,
+                     const CollisionRobot& other_robot, const robot_state::RobotState& other_state) const override;
+
+protected:
+  void updatedPaddingOrScaling(const std::vector<std::string>& links) override;
+  void addAttachedOjectsToManager(const robot_state::RobotState& state) const;
+  void getAttachedBodyObjects(const robot_state::AttachedBody* ab, std::vector<void*>& geoms) const;
+
+  void checkSelfCollisionHelper(const CollisionRequest& req, CollisionResult& res, const robot_state::RobotState& state,
+                                const AllowedCollisionMatrix* acm) const;
+  void checkOtherCollisionHelper(const CollisionRequest& req, CollisionResult& res,
+                                 const robot_state::RobotState& state, const CollisionRobot& other_robot,
+                                 const robot_state::RobotState& other_state, const AllowedCollisionMatrix* acm) const;
+
+  /** @brief Bullet collision manager taken from tesseract*/
+  mutable tesseract::tesseract_bullet::BulletDiscreteSimpleManager bt_manager_;
```

Comment:
Why is this mutable?

Reply:
In CollisionRobot, almost all functions are declared as const. When I now want to update the poses of the collision objects of the manager, I get an error because the functions through which CollisionRobot interacts are declared as `const`. However, I wanted to keep the CollisionRobot interface as unchanged as possible.

Reply:
I see. For each collision check in FCL a new manager is initialized and the full robot model is constructed from scratch. I probably wouldn't do it like this, either. Maybe we can keep a const instance of the manager already populated with the robot model and just clone it each time we need it? Cloning shouldn't be too costly since there is not much stored.